packages/file_selector/file_selector_web/integration_test/file_selector_web_test.dart

```diff
@@ -81,7 +80,17 @@ void main() {
   });
 }
 
-class MockDomHelper extends Mock implements DomHelper {}
+class MockDomHelper extends Mock implements DomHelper {
+  @override
+  Future<List<XFile>> getFiles({
+    String accept = '',
+    bool multiple = false,
+    FileUploadInputElement? input,
+  }) {
+    return super
+        .noSuchMethod(Invocation.method(#getFiles, [accept, multiple, input]));
+  }
```

Comment:
I'm confused by this override; it's not doing any of the nullability/non-nullability transformations described as necessary in mockito's NULL_SAFETY_README. Are you sure you were compiling and running the tests in strong mode?

(To clarify, when I referred to a manual mock I was actually thinking one that didn't use mockito at all, but instead just a simple class that implements `DomHelper` and has a method to set a return value for `getFiles` (and any argument checking to do). We could potentially do this too, but it doesn't look like this actually handles nullability correctly.)

Reply:
Ohh I thought I need to [manually mock](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md#manual-mock-implementaion) them.

Can you please share an already implemented such type of mock for reference?

Reply:
I don't think a different example would be useful given that it's class-specific. What I'm talking about is a class that implements the interface of the thing being mocked/faked, and also has whenever else is needed to populate it with fake data. For this, the simplest form would be something like (this is freeform so probably doesn't compile):

```
class FakeDomHelper implements DomHelper {
  List<XFile> filesToReturn = <XFile>[];

  @override
  Future<List<XFile>> getFiles({
    String accept = '',
    bool multiple = false,
    FileUploadInputElement? input,
  }) {
    return filesToReturn;
  }

  void setFiles(List<XFile> files) {
    filesToReturn = files;
  }
}
```

To keep the same level of correctness enforcement that the tests currently have you'd probably want to add `expectedAcceptStringParam` and `expectedMultipleParam` to that, and have `getFiles` return an empty array when there's no match. Or make the member variable a map of return values keyed by a combination of `accept` and `multiple`.

Reply:
> To keep the same level of correctness enforcement that the tests currently have you'd probably want to add `expectedAcceptStringParam` and `expectedMultipleParam` to that, and have `getFiles` return an empty array when there's no match. Or make the member variable a map of return values keyed by a combination of `accept` and `multiple`.

I couldn't get this but I have implemented it in a different way. Can you have a look at it?

Reply:
That's not implementing correctness enforcement a different way, it's ignoring it. This test does not validate that the calls are correct the way the previous test did.

Please use one of the approaches I described, or something equivalent, so that this isn't regressing test coverage. Alternately, @ditman, perhaps you could take over this portion of the PR?

Reply:
I've added expectation checks to the getFiles method. If the expected accept/multiple params don't match, the test will fail. Since we don't reuse the same mock twice, this is probably good enough.